EventGenerator/src/SingleProcessGenerator_module.cc

```diff
@@ -0,0 +1,143 @@
+// This module will be used to make electrons from any process originating from mu- stopped in any target material
+//
+// Sophie Middleton, 2021
+
+#include <iostream>
+#include <string>
+#include <cmath>
+#include <memory>
+#include <vector>
+
+#include "CLHEP/Vector/LorentzVector.h"
+#include "CLHEP/Random/RandomEngine.h"
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/RandExponential.h"
+#include "CLHEP/Units/PhysicalConstants.h"
+
+#include "fhiclcpp/types/Atom.h"
+#include "fhiclcpp/types/DelegatedParameter.h"
+#include "fhiclcpp/ParameterSet.h"
+
+#include "messagefacility/MessageLogger/MessageLogger.h"
+
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "art/Utilities/make_tool.h"
+
+#include "Offline/SeedService/inc/SeedService.hh"
+#include "Offline/GlobalConstantsService/inc/GlobalConstantsHandle.hh"
+#include "Offline/GlobalConstantsService/inc/ParticleDataTable.hh"
+#include "Offline/GlobalConstantsService/inc/PhysicsParams.hh"
+#include "Offline/DataProducts/inc/PDGCode.hh"
+#include "Offline/MCDataProducts/inc/StageParticle.hh"
+#include "Offline/Mu2eUtilities/inc/simParticleList.hh"
+#include "Offline/EventGenerator/inc/ParticleGeneratorTool.hh"
+
+namespace mu2e {
+  //================================================================
+  class SingleProcessGenerator : public art::EDProducer {
+  public:
+    struct Config {
+      using Name=fhicl::Name;
+      using Comment=fhicl::Comment;
+      fhicl::Atom<art::InputTag> inputSimParticles{Name("inputSimParticles"),
+          Comment("A SimParticleCollection with muons stopping in the target.")};
+      fhicl::Atom<std::string> stoppingTargetMaterial{
+        Name("stoppingTargetMaterial"),
+          Comment("Material determines muon life time, capture fraction, and particle spectra.\n"
```

Comment:
AFAIK this restriction applies only to LL DIO, as that has only been calculated for Al.  If other spectra are used this is (possibly) not a restriction, or a different restriction.  This is another situation where we need to tie BinnedSpectrum more tightly to the configuration.  That is outside the scope of this PR, I will add an issue.